File: teste01/trab.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#define N 1000

void imprime(float *lista, int size){
  int i;
  for (i=0;i<size;i++){
    printf("%f\n",lista[i]);
  }
  printf("%i linhas\n",size);
}
int ordena(float *lista, int size){
  int i;
  float temp;
  int troca=1;
  int passos=0;
  while(troca){
    troca=0;
    for(i=0;i<(size-1);i++){
      passos++;
      if (lista[i]>lista[i+1]){
	temp=lista[i];
	lista[i]=lista[i+1];
	lista[i+1]=temp;
	troca=1;
      }
    }
  }
  return passos;
}

int ordena2(float *lista, int size){
  int i;
  float temp;
  int troca=size-1;
  int newtroca;
  int passos=0;
  while(troca){
    for(i=0;i<troca;i++){
      passos++;
      if (lista[i]>lista[i+1]){
	temp=lista[i];
	lista[i]=lista[i+1];
	lista[i+1]=temp;
	newtroca=i;
      }
    }
    troca=newtroca;
  }
  return passos;
}

void main(){
  FILE *in;
  int i;
  float lista[N];
  float lista2[N];
  in=fopen("dados.txt","r");
  for(i=0; (!feof(in));i++){
    fscanf(in,"%f",&lista[i]);
    lista2[i]=lista[i];
  }
  imprime(lista,i);
  printf("passos: %i\n",ordena(lista,i));
  printf("passos: %i\n",ordena2(lista2,i));
}

File: teste01/trab.py
#!/usr/bin/python
# usar: python trab.py dados.txt
# ou    ./trab.py dados.txt

def ordena(lista1):
    lista=lista1[:]
    troca=True
    passos=0
    while(troca):
        troca=False
        for i in range(len(lista)-1):
            passos+=1
            if lista[i]>lista[i+1]:
                lista.insert(i,lista.pop(i+1))
                troca=True
    return (lista,passos)

def ordena2(lista1):
    lista=lista1[:]
    troca=True
    passos=0
    lasttroca=(len(lista)-1)
    while(troca):
        troca=False
        for i in range(lasttroca):
            passos+=1
            if lista[i]>lista[i+1]:
                lista.insert(i,lista.pop(i+1))
                troca=True
                lasttroca=i
    return (lista,passos)
        
def main(arquivo):
    with open(arquivo) as f:
        lines=f.readlines()
    numeros=[float(x) for x in lines]
    ord1=ordena(numeros)
    print 'ordena1',ord1[1]
    for x in ord1[0]:
        print x
    ord2=ordena2(numeros)
    print 'ordena2',ord2[1]
    for x in ord2[0]:
        print x
    
if __name__=="__main__":
    import sys
    main(*sys.argv[1:])
